### PR TITLE
Resolving issue #29 and #22.

### PR DESCRIPTION
**Pull request by juandlt-csu**
- For issue #29: added filters to upstream data pulls to only include the years 2022 and above. 
- For issue #22: 
    - Switched "where" column with "sonde_employed", and explained what this column does.
    - Split up `field_notes` and `deployment_record`. Removed padding in `deployment_record`.

These changes should close out issues #22 and #29.

### CODE REVIEW

File: src/qaqc/01-organize_raw_data.Rmd [hunk 1]
```diff
@@ -148,24 +149,34 @@ rm(cal_files, cal_tabler, package_loader)
...
   mutate(#start_DT = with_tz(start_DT, tzone = "MST"),
          DT_round = floor_date(start_DT, "15 minutes"),
          DT_join = as.character(DT_round),
          site = tolower(site),
-         season = year(DT_round)) %>%
+         season = year(DT_round))
+
+# Determining when the sonde was employed (SE) based on `sensor_pulled` (SP) and `sensor_deployed` (SD) columns.
+## when (SP & SD) columns are NOT empty, SE = 0 (employed).
+## when (SP) column is NOT empty, but (SD) column is empty, SE = 1 (NOT employed).
```
Comment (kathryn-willi) [inline src/qaqc/01-organize_raw_data.Rmd:163]: nice

File: src/qaqc/01-organize_raw_data.Rmd [hunk 2]
```diff
@@ -411,7 +422,7 @@ temperature_archery <- temperature_archery %>%
     flag),
          p1 = if_else(p1 >= temp_max | p1 <= temp_min, NA_real_, p1))
 
-# Here is where I would need to add the paste so that it doesn't replace what is already in the flag column completely
+# Here is sonde_employed I would need to add the paste so that it doesn't replace what is already in the flag column completely
```
Comment (kathryn-willi) [inline src/qaqc/01-organize_raw_data.Rmd:425]: LOL.... yeah just be sure to swap this out before next PR!

File: src/qaqc/01-organize_raw_data.Rmd [hunk 3]
```diff
@@ -337,21 +350,19 @@ summarize_site_param <- function(site_arg, parameter_arg, api_data) {
     mutate(DT_join = as.character(DT_round),
            site = site_arg,
            parameter = parameter_arg) %>% 
-    full_join(filter(dplyr::select(field_notes, where, visit_comments, sensor_malfunction, DT_join, site)),
-                 by = c('DT_join', 'site'))
+    full_join(filter(dplyr::select(deployment_record, sonde_employed, visit_comments, sensor_malfunction, DT_join, site)),
+                 by = c('DT_join', 'site')) %>% 
+    # determine if sonde was in the field or not
+    fill(sonde_employed)
```
Comment (kathryn-willi) [inline src/qaqc/01-organize_raw_data.Rmd:356]: Out of curiosity... how fast is this step now, compared to your old way?

**Review by kathryn-willi — APPROVED** [review]
Progress!!!

File: src/qaqc/01-organize_raw_data.Rmd (hunk 3 above)
Comment (juandlt-csu) [reply src/qaqc/01-organize_raw_data.Rmd:356]: The littlest bit faster, maybe. When I timed it the whole chunk takes about ~1 min to run (59.55sec). I feel like it took longer before.

**Pull request merged**